lib/snowplow-emr-etl-runner/contracts.rb

```diff
@@ -207,5 +208,12 @@ def Contract.failure_callback(data)
       :archive_enriched => Bool,
       :archive_shredded => Bool
     })
+
+    EmrStepConfig = ({
+      :step => Elasticity::JobFlowStep,
+      :retry_on_fail => Bool,
+      :rdb_loader_log => Maybe[[String, String]]
```

Comment:
Is there a difference between unset and empty list?

Reply:
Yes, an empty list is not valid and contracts will reject it.  It must be either

- A two-item list (effectively a tuple) containing a key and a s3 path.
- Unset

Translated into scala, this is `Option[(String, String)]`or Option[Tuple2[String, String]]`

Thanks a lot @chuwy for your patience with this one!  It's probably the hardest code change I did too.